Automattic/newspack-plugin
# feat(ras-acc): add segment control & build out button styles

### All Submissions:






### Changes proposed in this Pull Request:



_Note: This PR depends on the changes in https://github.com/Automattic/newspack-plugin/pull/3091 to be merged into it (like some class name changes), so I'm leaving it as a draft 'til then._ 

This PR adds the following block styles plus examples of each:

* It adds the small and x-small button styles for the regular buttons (medium is the default).
* It adds the small and medium button styles for icon buttons (x-small is the default). 
* It adds examples of buttons with icons, and tweaks the spacing.
* It adds the 'segment component' which kind of uses button styles combined with tabbed behaviour. 
* It adds a loading style for the button.

### How to test the changes in this Pull Request:

1. Apply the PR and run `npm run build`.
2. Review the X-Small and Small button examples, which include icons -- these appear after the 'wide' buttons. Because it's the most commonly used size, the 'Medium' size is the default for the regular buttons.

![image](https://github.com/Automattic/newspack-plugin/assets/177561/8c3b9615-8a1a-49e7-b6dc-f06891ae971a)

3. Review the Loading button examples -- these appear next. The animation is added through CSS, but the markup relies on a `<span>` being added around the button's text -- open to feedback on this specific point!

![image](https://github.com/Automattic/newspack-plugin/assets/177561/62d24d42-9391-4fec-bf5c-320d3d90163b)

4. Review the Segment Controls -- there should be two examples, one with actual tab content, and one that just acts like free-standing buttons:

![image](https://github.com/Automattic/newspack-plugin/assets/177561/037af7a4-22c2-45dd-8de4-191ef8914a01)

At the very bottom, there's also an example of what this component looks like inside of a modal (which better matches how it's used in the My Account mockups):

![image](https://github.com/Automattic/newspack-plugin/assets/177561/a8d0cf1d-9f55-4a18-befc-8eb46a36b843)

5. Lastly, review the icon buttons -- a new Small and Medium size have been added. The regular buttons use 'Medium' as a default, but the icon button uses X-Small -- for both, this is because they're the most-used size, but it may be unnecessarily confusing. I'd appreciate some any feedback around how best to balance the defaults, to make sure things make sense and are easy to use:

![image](https://github.com/Automattic/newspack-plugin/assets/177561/efd03dd7-0b48-4f59-95c4-1f470835fe7b)

### Other information:
